// carbon-components-angular/ui-shell/header/header-action.component.ngfactory.js
/**
 *
 * carbon-angular v0.0.0 | header-action.component.ngfactory.js
 *
 * Copyright 2014, 2020 IBM
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0

 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */


/**
 * @fileoverview This file was generated by the Angular template compiler. Do not edit.
 *
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride,checkTypes}
 * tslint:disable
 */ 
import * as i0 from "@angular/core";
import * as i1 from "@carbon/icons-angular/lib/close/20";
import * as i2 from "@angular/common";
import * as i3 from "./header-action.component";
var styles_HeaderAction = [];
var RenderType_HeaderAction = i0.ɵcrt({ encapsulation: 2, styles: styles_HeaderAction, data: {} });
export { RenderType_HeaderAction as RenderType_HeaderAction };
function View_HeaderAction_1(_l) { return i0.ɵvid(0, [i0.ɵncd(null, 0), (_l()(), i0.ɵand(0, null, null, 0))], null, null); }
function View_HeaderAction_2(_l) { return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 1, ":svg:svg", [["ibmIconClose20", ""]], null, null, null, null, null)), i0.ɵdid(1, 4210688, null, 0, i1.Close20Directive, [i0.ElementRef], null, null)], null, null); }
export function View_HeaderAction_0(_l) { return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 6, "button", [["class", "bx--header__action"]], [[1, "aria-label", 0], [8, "title", 0]], [[null, "click"]], function (_v, en, $event) { var ad = true; var _co = _v.component; if (("click" === en)) {
        var pd_0 = (_co.onClick() !== false);
        ad = (pd_0 && ad);
    } return ad; }, null, null)), i0.ɵdid(1, 278528, null, 0, i2.NgClass, [i0.IterableDiffers, i0.KeyValueDiffers, i0.ElementRef, i0.Renderer2], { klass: [0, "klass"], ngClass: [1, "ngClass"] }, null), i0.ɵpod(2, { "bx--header__action--active": 0 }), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_HeaderAction_1)), i0.ɵdid(4, 16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, "ngIf"] }, null), (_l()(), i0.ɵand(16777216, null, null, 1, null, View_HeaderAction_2)), i0.ɵdid(6, 16384, null, 0, i2.NgIf, [i0.ViewContainerRef, i0.TemplateRef], { ngIf: [0, "ngIf"] }, null)], function (_ck, _v) { var _co = _v.component; var currVal_2 = "bx--header__action"; var currVal_3 = _ck(_v, 2, 0, _co.active); _ck(_v, 1, 0, currVal_2, currVal_3); var currVal_4 = !_co.active; _ck(_v, 4, 0, currVal_4); var currVal_5 = _co.active; _ck(_v, 6, 0, currVal_5); }, function (_ck, _v) { var _co = _v.component; var currVal_0 = _co.title; var currVal_1 = _co.title; _ck(_v, 0, 0, currVal_0, currVal_1); }); }
export function View_HeaderAction_Host_0(_l) { return i0.ɵvid(0, [(_l()(), i0.ɵeld(0, 0, null, null, 1, "ibm-header-action", [], null, null, null, View_HeaderAction_0, RenderType_HeaderAction)), i0.ɵdid(1, 49152, null, 0, i3.HeaderAction, [], null, null)], null, null); }
var HeaderActionNgFactory = i0.ɵccf("ibm-header-action", i3.HeaderAction, View_HeaderAction_Host_0, { title: "title", active: "active", activeChange: "activeChange" }, { selected: "selected" }, ["*"]);
export { HeaderActionNgFactory as HeaderActionNgFactory };
